### ISSUE TITLE
Prevent package when included with plugins implementing CEF

### ISSUE THREAD

**Issue by MaximeM** [millicast/millicast-publisher-unreal-engine-plugin#26]
Hello,

I can't get to package this plugin( 5.1 branch), with any plugin integrating CEF used for web views.

Such as [BLUI Unreal](https://github.com/getnamo/BLUI-Unreal/)

Always start with the same callstack:

UATHelper: Packaging (Windows): LogWindows: Error: Fatal error!
UATHelper: Packaging (Windows): LogWindows: Error:
UATHelper: Packaging (Windows): LogWindows: Error: Unhandled Exception: 0x80000003
UATHelper: Packaging (Windows): LogWindows: Error:
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ffc7661a5c5 libcef.dll!UnknownFunction []

Any clues ?

Thanks for your swift and precise answers everytime.

Max

**Comment by dbaldassi** [comment]
Hi, I never work with a CEF plugin before.
Is the plugin embedding WebRTC ? there might be some conflicts causing this runtime issue.
When does this error occur ? Just launching the game in the editor ?

**Comment by MaximeM** [comment]
It's only causing problem at package

**Comment by rweber89** [comment]
Hello @MaximeM,

That define is automatically generated. Our plugin isn't using CEF anywhere. However I have been able to both reproduce and fix the issue.

Our plugin does capture audio, which was making CEF initialization fail. Once this PR is merged into the dev branch please check further on your end: https://github.com/millicast/millicast-publisher-unreal-engine-plugin/pull/42

It is possible that there will be issues once your project is running, however if it works during Play-in-Editor, then it should also work as a packaged build.

Please do update us!

**Comment by MaximeM** [comment]
Hey thanks for working on it, it's weird I have this crash upon level opening
```
UnrealEditor_AudioCaptureRtAudio
UnrealEditor_AudioCaptureRtAudio
UnrealEditor_AudioCaptureRtAudio
UnrealEditor_AudioCaptureCore
UnrealEditor_MillicastPublisher!Millicast::Publisher::AudioDeviceCapturer::GetCaptureDevicesAvailable() [E:\Powerz\P4_Powerz\Omega_Park\Plugins\MillicastPublisher\Source\MillicastPublisher\Private\Media\AudioDeviceCapturer.cpp:107]
UnrealEditor_MillicastPublisher!UMillicastPublisherSource::RefreshAudioDevicesList() [E:\Powerz\P4_Powerz\Omega_Park\Plugins\MillicastPublisher\Source\MillicastPublisher\Private\Media\MillicastPublisherSource.cpp:121]
UnrealEditor_MillicastPublisher!UMillicastPublisherSource::UMillicastPublisherSource() [E:\Powerz\P4_Powerz\Omega_Park\Plugins\MillicastPublisher\Source\MillicastPublisher\Private\Media\MillicastPublisherSource.cpp:22]

```

**Comment by MaximeM** [comment]
And this callstack when packaging with just millicast player and publisher

```
ATHelper: Packaging (Windows): LogWindows: Error: begin: stack for UAT
UATHelper: Packaging (Windows): LogWindows: Error: === Critical error: ===
UATHelper: Packaging (Windows): LogWindows: Error:
UATHelper: Packaging (Windows): LogWindows: Error: Fatal error!
UATHelper: Packaging (Windows): LogWindows: Error:
UATHelper: Packaging (Windows): LogWindows: Error: Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0xffffffffffffffff
UATHelper: Packaging (Windows): LogWindows: Error:
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ffa6b251b70 msvcrt.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff9ea6f86bf DSOUND.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff9ea702e4f DSOUND.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff9ea7210fa DSOUND.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff9ea710f2a DSOUND.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff939189b0e UnrealEditor-AudioCaptureRtAudio.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff939184814 UnrealEditor-AudioCaptureRtAudio.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff939184b26 UnrealEditor-AudioCaptureRtAudio.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff9710421f2 UnrealEditor-AudioCaptureCore.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff96734be0f UnrealEditor-MillicastPublisher.dll!Millicast::Publisher::AudioDeviceCapturer::GetCaptureDevicesAvailable() [E:\Powerz\P4_Powerz\Omega_Park\Plugins\MillicastPublisher\Source\MillicastPublisher\Private\Media\AudioDeviceCapturer.cpp:107]
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff96735682d UnrealEditor-MillicastPublisher.dll!UMillicastPublisherSource::RefreshAudioDevicesList() [E:\Powerz\P4_Powerz\Omega_Park\Plugins\MillicastPublisher\Source\MillicastPublisher\Private\Media\MillicastPublisherSource.cpp:121]
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff967336704 UnrealEditor-MillicastPublisher.dll!UMillicastPublisherSource::UMillicastPublisherSource() [E:\Powerz\P4_Powerz\Omega_Park\Plugins\MillicastPublisher\Source\MillicastPublisher\Private\Media\MillicastPublisherSource.cpp:22]
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff9a0025387 UnrealEditor-CoreUObject.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff99fe49c26 UnrealEditor-CoreUObject.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff99fe4a4bb UnrealEditor-CoreUObject.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff99fe73761 UnrealEditor-CoreUObject.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff9a000d9eb UnrealEditor-CoreUObject.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff9a000c3c4 UnrealEditor-CoreUObject.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff9a000c930 UnrealEditor-CoreUObject.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff98dc657d2 UnrealEditor-UnrealEd.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff98dc65a02 UnrealEditor-UnrealEd.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff98dc6f810 UnrealEditor-UnrealEd.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff98dc81870 UnrealEditor-UnrealEd.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff98dc811a5 UnrealEditor-UnrealEd.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff98d9df780 UnrealEditor-UnrealEd.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff98da016a3 UnrealEditor-UnrealEd.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff77231b0ce UnrealEditor-Cmd.exe!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff772312688 UnrealEditor-Cmd.exe!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff77231296a UnrealEditor-Cmd.exe!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff772315680 UnrealEditor-Cmd.exe!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff7723271b4 UnrealEditor-Cmd.exe!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ff7723296f2 UnrealEditor-Cmd.exe!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ffa69bc26bd KERNEL32.DLL!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error: [Callstack] 0x00007ffa6b80dfb8 ntdll.dll!UnknownFunction []
UATHelper: Packaging (Windows): LogWindows: Error:
UATHelper: Packaging (Windows): LogWindows: Error: end: stack for UAT
PackagingResults: Error: begin: stack for UAT
PackagingResults: Error: === Critical error: ===
PackagingResults: Error: Fatal error!
PackagingResults: Error: Unhandled Exception: EXCEPTION_ACCESS_VIOLATION reading address 0xffffffffffffffff
PackagingResults: Error: [Callstack] 0x00007ffa6b251b70 msvcrt.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff9ea6f86bf DSOUND.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff9ea702e4f DSOUND.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff9ea7210fa DSOUND.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff9ea710f2a DSOUND.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff939189b0e UnrealEditor-AudioCaptureRtAudio.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff939184814 UnrealEditor-AudioCaptureRtAudio.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff939184b26 UnrealEditor-AudioCaptureRtAudio.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff9710421f2 UnrealEditor-AudioCaptureCore.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff96734be0f UnrealEditor-MillicastPublisher.dll!Millicast::Publisher::AudioDeviceCapturer::GetCaptureDevicesAvailable() [E:\Powerz\P4_Powerz\Omega_Park\Plugins\MillicastPublisher\Source\MillicastPublisher\Private\Media\AudioDeviceCapturer.cpp:107]
PackagingResults: Error: [Callstack] 0x00007ff96735682d UnrealEditor-MillicastPublisher.dll!UMillicastPublisherSource::RefreshAudioDevicesList() [E:\Powerz\P4_Powerz\Omega_Park\Plugins\MillicastPublisher\Source\MillicastPublisher\Private\Media\MillicastPublisherSource.cpp:121]
PackagingResults: Error: [Callstack] 0x00007ff967336704 UnrealEditor-MillicastPublisher.dll!UMillicastPublisherSource::UMillicastPublisherSource() [E:\Powerz\P4_Powerz\Omega_Park\Plugins\MillicastPublisher\Source\MillicastPublisher\Private\Media\MillicastPublisherSource.cpp:22]
PackagingResults: Error: [Callstack] 0x00007ff9a0025387 UnrealEditor-CoreUObject.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff99fe49c26 UnrealEditor-CoreUObject.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff99fe4a4bb UnrealEditor-CoreUObject.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff99fe73761 UnrealEditor-CoreUObject.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff9a000d9eb UnrealEditor-CoreUObject.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff9a000c3c4 UnrealEditor-CoreUObject.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff9a000c930 UnrealEditor-CoreUObject.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff98dc657d2 UnrealEditor-UnrealEd.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff98dc65a02 UnrealEditor-UnrealEd.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff98dc6f810 UnrealEditor-UnrealEd.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff98dc81870 UnrealEditor-UnrealEd.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff98dc811a5 UnrealEditor-UnrealEd.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff98d9df780 UnrealEditor-UnrealEd.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff98da016a3 UnrealEditor-UnrealEd.dll!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff77231b0ce UnrealEditor-Cmd.exe!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff772312688 UnrealEditor-Cmd.exe!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff77231296a UnrealEditor-Cmd.exe!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff772315680 UnrealEditor-Cmd.exe!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff7723271b4 UnrealEditor-Cmd.exe!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ff7723296f2 UnrealEditor-Cmd.exe!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ffa69bc26bd KERNEL32.DLL!UnknownFunction []
PackagingResults: Error: [Callstack] 0x00007ffa6b80dfb8 ntdll.dll!UnknownFunction []
PackagingResults: Error: end: stack for UAT
```

**Comment by rweber89** [comment]
Thanks for the new callstack. I'll provide an update later.

**Comment by rweber89** [comment]
@MaximeM If you would like to check out the changes before they are merged to see if it helps, there is a branch for this PR:
https://github.com/millicast/millicast-publisher-unreal-engine-plugin/pull/46

Please let me know if that resolves it for you, as I cannot reproduce the crash locally.
